pkg/controllers/terminating/v1alpha1/terminate.go

```diff
@@ -38,110 +37,72 @@ type Terminator struct {
 	coreV1Client  corev1.CoreV1Interface
 }
 
-// cordonNodes takes in a list of expired nodes as input and cordons them
-func (t *Terminator) cordonNodes(ctx context.Context) error {
-	// 1. Get terminable nodes
-	nodeList, err := t.getLabeledNodes(ctx, provisioning.ProvisionerTerminablePhase)
-	if err != nil {
-		return err
+// cordonNode takes in a list of expired nodes as input and cordons them
+func (t *Terminator) cordonNode(ctx context.Context, node *v1.Node) error {
+	if node.Spec.Unschedulable {
+		return nil
 	}
-	// 2. Cordon nodes
-	for _, node := range nodeList {
-		persisted := node.DeepCopy()
-		node.Spec.Unschedulable = true
-		node.Labels = functional.UnionStringMaps(
-			node.Labels,
-			map[string]string{provisioning.ProvisionerPhaseLabel: provisioning.ProvisionerDrainingPhase},
-		)
-		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
-			return fmt.Errorf("patching node %s, %w", node.Name, err)
-		}
-		zap.S().Debugf("Cordoned node %s", node.Name)
+	persisted := node.DeepCopy()
+	node.Spec.Unschedulable = true
+	node.Labels = functional.UnionStringMaps(
```

Comment:
How about

`node.Labels[provisioning.ProvisionerPhaseLabel] = provisioning.ProvisionerDrainingPhase`

Reply:
That does the trick. Thanks.